src/scenes/lobbyscreen.ts

```diff
@@ -0,0 +1,148 @@
+        this.me = SessionManager.getSessionInformation();
+    }
+
+    public init(): void {
+        const playerManagerListener: UUID = PlayerManager.onSync.on(this.syncPlayers);
+        const sessionManagerListener: UUID = SessionManager.onSync.on(this.syncMe);
+
+        // on scene destroy free listener
+        this.events.on("destroy", () => {
+            PlayerManager.onSync.off(playerManagerListener);
+            SessionManager.onSync.off(sessionManagerListener);
+        });
+    }
+    
+
+    public preload(): void{
+        this.load.image("testButton", "assets/testButton.png");
+    }
+
+    graphics!: Phaser.GameObjects.Graphics;
+
+    public create(): void {
+        this.text = this.add
+            .text(350, 80, "Saboteur Lobby", {
+                fontFamily: "Goudy",
+                fontSize: 38,
+                color: "#ffffff",
+                stroke: "#000000",
+                strokeThickness: 8,
+                align: "center",
+            })
+            .setOrigin(0.5);
+        log(this, this.text);
+
+        
+        let demoButton = this.add.rectangle(400, 520, 150, 50, 0x5C4033);
+        demoButton.setStrokeStyle(2, 0x000000,);
+
+        // Add text within the rectangle
+        const buttonText = this.add.text(0, 0, "Click me", { 
+            fontFamily: "Arial", 
+            fontSize: "20px", 
+            color: "#FFFFFF",
+        });
+
+        Phaser.Display.Align.In.Center(buttonText, demoButton);
+
+        // Make the text interactive
+        demoButton.setInteractive();
+        // Add a pointerdown event listener to the text
+        demoButton.on("pointerdown", () => {
+            log("click me!");
+            // Add your custom functionality here
+        });
+        demoButton.on("pointerover", () => {
+            this.game.canvas.style.cursor = "pointer";
+        });
+        demoButton.on("pointerout", () => {
+            this.game.canvas.style.cursor = "default";
+        });
+
+        const startButton = this.add.image(150, 540, "testButton");
+        startButton.setInteractive();
+        startButton.on("pointerdown", () => {
+            log("Click brown button"); // Call a custom function when the button is clicked
+        });
+        startButton.on("pointerover", () => {
+            this.game.canvas.style.cursor = "pointer";
+        });
+        startButton.on("pointerout", () => {
+            this.game.canvas.style.cursor = "default";
+        });
+
+        const createText = this.add.text(580, 540, "Start", { 
+            color: "#ffffff", 
+            fontSize: "50px",
+        });
+        createText.setInteractive();
+        createText.on("pointerdown", () => {
+            log("Click on start text");
+        });
+        createText.on("pointerover", () => {
+            this.game.canvas.style.cursor = "pointer";
+        });
+        createText.on("pointerout", () => {
+            this.game.canvas.style.cursor = "default";
+        });
+
+    }
```

Comment:
please add variable return type everywhere and get the interactivity function from @Croxsy and use this to make things interactive.

Reply:
Cant add the type "string" to this.game.canvas.style.cursor without TypeScript complaining, it should be a string that holds a CSS value.